AsyncDisplayKit/ASCellNode.m

```diff
@@ -132,7 +122,7 @@ - (void)setText:(NSString *)text
   _textNode.attributedString = [[NSAttributedString alloc] initWithString:_text
                                                                attributes:@{NSFontAttributeName: [UIFont systemFontOfSize:kFontSize]}];
 
-  [self invalidateCalculatedSize];
+  [self invalidateCalculatedLayout];
```

Comment:
What do you think about naming this -invalidateLayout?  I would suggest this change later, but anyone who updates to master (a fair number of ASDK clients build against master) would have to change it in their code again.

In thinking about this, consider what we will need to do as the very most important change after this one (even better if we can do it now): more gracefully handling re-layouts, like what happens today if you rotate the Kittens app (text does not reflow, new cells are at the full width but also do not re-wrap when rotating back to portrait).

If that were fixed, we'd need to understand the expected behavior in handling asynchronous layout; fortunately, ASDataController already does this in batches, so if we passed through ASTableView / ASCollectionView -layoutSubviews calls as triggering cell-relayout, things would probably show up at once.  We shouldn't do anything specific for scroll point restoration as the cells change height.

With those changes, I suspect we would make -setNeedsLayout do the bulk of this work: doing that would kick off an asynchronous measurement pass and then automatically trigger -layout afterwards.  I'm not totally sure, but think we might be able to get rid of -invalidateLayout entirely...which would be a huge success enabled by your work, as -invalidateCalculatedSize has always been one more custom API for developers to try to understand.